File: frontend/src/objects/merchant/store/UpdateStoreOperationalRequest.ts
import type { BusinessHours } from '@/objects/merchant/store/BusinessHours'
import type { AddressText, DeliveryCoordinate, Minutes } from '@/objects/domain/DomainObjects'

export type UpdateStoreOperationalRequest = {
  storeAddress: AddressText
  location?: DeliveryCoordinate
  businessHours: BusinessHours
  avgPrepMinutes: Minutes
}

File: backend/src/main/scala/merchant/object/store/UpdateStoreOperationalRequest.scala
package domain.merchant

import domain.shared.given

import domain.shared.*
import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.*

final case class UpdateStoreOperationalRequest(
    storeAddress: AddressText,
    businessHours: BusinessHours,
    avgPrepMinutes: Minutes,
)
object UpdateStoreOperationalRequest:
  given Encoder[UpdateStoreOperationalRequest] = deriveEncoder
  given Decoder[UpdateStoreOperationalRequest] = deriveDecoder
